### PR TITLE
fix: `Required` option not handled correctly for special fields (File, GeoPoint, Polygon) on GraphQL API mutations

### LINKED ISSUE #8914: required field on Parse Schema is not handled correctly on special types on the GQL API
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description

When a special field like File is required on the Parse Schema, mutations fails because the GQL type is wrapped by a GraphQL Non Null type.

### Steps to reproduce

Add a "required: true" to a special field like a field with a type "File", and attempt a mutation

### Actual Outcome

Mutation fails, Parse fail to recognize correctly the special field.

### Environment

Server
- Parse Server version: `7.0.0`
- Operating system: `MacOS`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `Local`

Database
- System (MongoDB or Postgres): `Mongo`
- Database version: `6.0.2`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `FILL_THIS_OUT`

Client
- SDK (iOS, Android, JavaScript, PHP, Unity, etc): `FILL_THIS_OUT`
- SDK version: `FILL_THIS_OUT`

### PR DESCRIPTION
**Pull request by Moumouls**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

## Issue

Closes: #8914

## Approach

Base all field detection on schema data ,and not on the GQL type ( like the Relation and Pointer type)

## Tasks

- [x] Add tests
- [x] Add changes to documentation (guides, repository pages, code comments)
- [x] Add [security check](https://github.com/parse-community/parse-server/blob/master/CONTRIBUTING.md#security-checks)
- [x] Add new Parse Error codes to Parse JS SDK

### CODE REVIEW

**Comment by mtrezza** [comment]
Is this a breaking change?

**Comment by mtrezza** [comment]
@Moumouls Could you please resolve the conflicts, and clarify whether this is a breaking change intended for Parse Server 7?  In a few days we'll release Parse Server 7, so if this is a breaking change, merging it before release would be easier.

File: spec/ParseGraphQLServer.spec.js [hunk 1]
```diff
@@ -88,8 +88,8 @@ describe('ParseGraphQLServer', () => {
 
     it('should initialize parseGraphQLSchema with a log controller', async () => {
       const loggerAdapter = {
-        log: () => {},
-        error: () => {},
+        log: () => { },
```
Comment (mtrezza) [inline spec/ParseGraphQLServer.spec.js]: Could you remove these spaces inside the brackets? Seems to be added by your local lint?

**Review by mtrezza — CHANGES_REQUESTED** [review]
Requires lint fixes; it seems that the local lint adds spaces in curly braces and changes indentation.

**Review by mtrezza — APPROVED** [review]

File: spec/ParseGraphQLServer.spec.js (hunk 1 above)
Comment (mtrezza) [reply spec/ParseGraphQLServer.spec.js]: fixed lint errors

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.0.0-alpha.30](https://github.com/parse-community/parse-server/releases/tag/7.0.0-alpha.30)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.1.0-alpha.1](https://github.com/parse-community/parse-server/releases/tag/7.1.0-alpha.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.1.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/7.1.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.1.0](https://github.com/parse-community/parse-server/releases/tag/7.1.0)

**Pull request merged**